### PR TITLE
c2c: skip and deflake a few unit tests

### LINKED ISSUE #102523: ccl/streamingccl/streamproducer: TestStreamAddSSTable failed
ccl/streamingccl/streamproducer.TestStreamAddSSTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9849993?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9849993?buildTab=artifacts#/) on master @ [3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f](https://github.com/cockroachdb/cockroach/commits/3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f):

```
=== RUN   TestStreamAddSSTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9ce24496530642208197dda818d65c9e/logTestStreamAddSSTable910410699
    test_log_scope.go:79: use -show-logs to present logs inline
    replication_stream_test.go:563: -- test log scope end --
    replication_stream_test.go:563: Leaked goroutine: goroutine 5947038 [select]:
        github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch.func1()
        	github.com/jackc/pgconn/internal/ctxwatch/external/com_github_jackc_pgconn/internal/ctxwatch/context_watcher.go:47 +0x6c
        created by github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch
        	github.com/jackc/pgconn/internal/ctxwatch/external/com_github_jackc_pgconn/internal/ctxwatch/context_watcher.go:46 +0x12c
--- FAIL: TestStreamAddSSTable (26.09s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-27490

### PR DESCRIPTION
**Pull request by msbutler**
c2c: skip a few unit tests
Informs #102523
Informs #102286
Informs #86206

Release note: none

c2c: close pgconn with correct context in test infra
When a user opens a pgx connection, they pass it a context that must be used to
close the connection, else pgx's contextWatcher will run indefinetly on a
seperate goroutine.  Previously, the `startReplication()` test helper was
initializing its own context to open pgx connection, and did not pass this
context back to the caller, and consequently, the user would close the
connection with the incorrect context, leading to a leaky goroutine.

This patch allows the user to pass a context in the startReplication(),
preventing the user from closing the connection with the wrong goroutine.

Fixes https://github.com/cockroachdb/cockroach/issues/102523

Release note: None

### CODE REVIEW

**Review by benbardin — APPROVED** [review]

File: pkg/util/leaktest/leaktest.go [hunk 1]
```diff
@@ -58,8 +58,9 @@ func interestingGoroutines() map[int64]string {
 			strings.Contains(stack, "sentry-go.(*HTTPTransport).worker") ||
 			// Ignore the opensensus worker, which is created by the event exporter.
 			strings.Contains(stack, "go.opencensus.io/stats/view.(*worker).start") ||
-			// Ignore pgconn which creates a goroutine to do an async cleanup.
+			// Ignore pgconn which creates two goroutines to do async cleanup.
 			strings.Contains(stack, "github.com/jackc/pgconn.(*PgConn).asyncClose.func1") ||
+			strings.Contains(stack, "github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch.func1()") ||
```
Comment (adityamaru) [inline pkg/util/leaktest/leaktest.go:63]: what is this about? just want to check its intentional

Comment (dt) [reply pkg/util/leaktest/leaktest.go:63]: we should be very sure this isn't a missing Close() before we add it to the list or we could be hiding a real drain / shutdown bug

Comment (msbutler) [reply pkg/util/leaktest/leaktest.go]: good call. i'll investigate further and skip the flaky test for now.

**Comment by msbutler** [comment]
@dt I think I found the test infra bug. I think we were mishandling contexts. I wonder if I should revert https://github.com/cockroachdb/cockroach/pull/102258 as well?

**Comment by msbutler** [comment]
bors r=benbardin

**Comment by msbutler** [comment]
bors retry

**Comment by msbutler** [comment]
blathers backport 23.1

**Pull request merged**